tests/unit/test_bucket.py

```diff
@@ -88,24 +88,69 @@ def test_ctor_w_number_of_newer_versions(self):
         self.assertIsNone(conditions.matches_storage_class)
         self.assertEqual(conditions.number_of_newer_versions, 3)
 
+    def test_ctor_w_days_since_noncurrent_time(self):
+        conditions = self._make_one(
+            number_of_newer_versions=3, days_since_noncurrent_time=2
+        )
+        expected = {"numNewerVersions": 3, "daysSinceNoncurrentTime": 2}
+        self.assertEqual(dict(conditions), expected)
+        self.assertIsNone(conditions.age)
+        self.assertIsNone(conditions.created_before)
+        self.assertIsNone(conditions.is_live)
+        self.assertIsNone(conditions.matches_storage_class)
+        self.assertEqual(conditions.number_of_newer_versions, 3)
+        self.assertEqual(conditions.days_since_noncurrent_time, 2)
+
+    def test_ctor_w_noncurrent_time_before(self):
+        import datetime
+        from google.cloud._helpers import _datetime_to_rfc3339
+
+        noncurrent_before = datetime.datetime.now() + datetime.timedelta(days=10)
+        conditions = self._make_one(
+            number_of_newer_versions=3, noncurrent_time_before=noncurrent_before
+        )
+
+        expected = {
+            "numNewerVersions": 3,
+            "noncurrentTimeBefore": _datetime_to_rfc3339(noncurrent_before),
+        }
+        self.assertEqual(dict(conditions), expected)
+        self.assertIsNone(conditions.age)
+        self.assertIsNone(conditions.created_before)
+        self.assertIsNone(conditions.is_live)
+        self.assertIsNone(conditions.matches_storage_class)
+        self.assertEqual(conditions.number_of_newer_versions, 3)
+        self.assertEqual(
+            conditions.noncurrent_time_before, _datetime_to_rfc3339(noncurrent_before)
```

Comment:
As above, ISTM that the property should return `datetime`, rather than the RFC string.